addons/sale/static/src/js/reconciliation/reconciliation_action.js

```diff
@@ -0,0 +1,285 @@
+odoo.define('sale.ReconciliationClientAction', function (require) {
+"use strict";
+
+var ReconciliationClientAction = require('account.ReconciliationClientAction');
+var core = require('web.core');
+var fieldUtils = require('web.field_utils');
+var Dialog = require('web.view_dialogs');
+
+var _t = core._t;
+var saleOrderSelectViewId;
+var saleOrderSearchViewId;
+
+
+ReconciliationClientAction.StatementAction.include({
+    custom_events: _.extend({}, ReconciliationClientAction.StatementAction.prototype.custom_events, {
+        reconcile_with_sale_order: '_onReconcileWithSaleOrder',
+    }),
+
+    /**
+     * @override
+     */
+    willStart: function () {
+        return $.when(this._super.apply(this, arguments),
```

Comment:
var selectViewDef = !saleOrderSelectViewId && this._rpc(...);
var searchViewDef = !saleOrderSearchViewId && this._rpc(...);

return $.when(this._super.apply(this, arguments), selectViewDef, searchViewDef);